server/bin/pbench-check-tb-age.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""Pbench Check TB Age
+
+Return 0 if the given tar ball (full path) is within the configured maximum
+age (specified in days), and return 1 if it is not.
+
+A return value > 1 indicates some other error with the command.
+"""
+
+import sys
+import os
+import re
+from datetime import datetime
+from argparse import ArgumentParser
+from configparser import NoOptionError
+
+from pbench import (
+    PbenchConfig,
+    BadConfig,
+)
+
+
+_NAME_ = "pbench-check-tb_age"
+
+tb_pat_r = (
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+tb_pat = re.compile(tb_pat_r)
+
+
+def main(options):
+    if not options.tb_path:
+        print(
+            f"{_NAME_}: ERROR: No tar ball path specified", file=sys.stderr,
+        )
+        return 2
```

Comment:
It would be nice if the return codes are documented.

Reply:
The emitted error is the documentation for the return code.  We could add a function documentation string, but then we'd be duplicating the strings in the error messages.

What do you suggest?

Reply:
Still not sure what end will that serve.  Can you provide a scenario for that?  Usually, like errno values, you keep the association of number to error string independent of the code using them, so that more than one piece of code can use the same numbers without having to duplicate the strings.

In this case, there is only one place where these codes have meaning.  The error codes do get used in the `INFO` log message produced by `pbench-unpack-tarballs.sh`.  In that case, the consumer of the codes is the human being reading the log output, to which they also have access to the error message emitted by `pbench-check-tb-age`.

Let's discuss further if there is another situation that needs to be considered.